cmd/main_test.go

```diff
@@ -301,3 +307,78 @@ var _ = DescribeTable("TestRunAccessCheck", func(user string, namespace string,
 		"patch",
 		false),
 )
+
+var _ = DescribeTable("TestGetNamespacesWithAccess", func(allNamespaces []k8sapi.Namespace,
+	expectedNs []k8sapi.Namespace, actualNs []k8sapi.Namespace, err error) {
+	e := echo.New()
+	cfg, _ := config.GetConfig()
+	clientset, _ := kubernetes.NewForConfig(cfg)
+	req := httptest.NewRequest(http.MethodGet, "/", nil)
+	rec := httptest.NewRecorder()
+	c := e.NewContext(req, rec)
+	c.Request().Header.Set("X-Email", "user1@konflux.dev")
+	authCl := clientset.AuthorizationV1()
+	
+	JustBeforeEach(func() {
+		actualNs, err = getNamespacesWithAccess(e, c, authCl, allNamespaces)
+	})
+
+	Context("When namespaces ns1 and ns2 has all the necessary permissions like create, list, watch and delete", func() {
+		BeforeEach(func() {
+			namespaceNames := []string{"test-tenant", "test-tenant-2"}
+			for _, name := range namespaceNames {
+				ns, err := createNamespace(k8sClient, name)
+				Expect(err).NotTo(HaveOccurred(), fmt.Sprintf("Error while creating the namespace %s: %v", name, err))
+				allNamespaces = append(allNamespaces, ns)
+			}			
+			createRole(k8sClient, "test-tenant", "namespace-access", []string{"create", "list", "watch", "delete"})
+			createRole(k8sClient, "test-tenant-2", "namespace-access-2", []string{"create", "list", "watch", "delete"})
+			createRoleBinding(k8sClient, "namespace-access-user-binding", "test-tenant", "user1@konflux.dev", "namespace-access")
+			createRoleBinding(k8sClient, "namespace-access-user-binding-2", "test-tenant", "user2@konflux.dev", "namespace-access")
+			expectedNs = allNamespaces
+		})
+		It("returns all namespaces ns1 and ns2", func() {
+			Expect(actualNs).To(Equal(expectedNs))
+			Expect(err).NotTo(HaveOccurred(), "Unexpected error testing GetNamespacesWithAccess")
+		})
+	})
+
+	Context("When namspace ns3 doesn't have necessary permissions", func() {
+		BeforeEach(func() {
+			var ns3 k8sapi.Namespace
+			ns3, err = createNamespace(k8sClient, "test-tenant-3")
+			createRoleBinding(k8sClient, "namespace-access-user-binding-3", "test-tenant-2", "user2@konflux.dev", "namespace-access-2")
+			Expect(err).NotTo(HaveOccurred(), fmt.Sprintf("Error while creating the namespace ns3: %v", err))
+			allNamespaces = []k8sapi.Namespace{ns3}
+			expectedNs = []k8sapi.Namespace{}
+		})
+		It("doesn't return any namespace", func() {
+			Expect(actualNs).To(Equal(expectedNs))
+			Expect(err).NotTo(HaveOccurred(), "Unexpected error testing GetNamespacesWithAccess")
+		})
+	})
+
+	Context("When namespaces ns1 and ns2 has all necessary permissions and ns3, ns4 doesn't", func() {
+		BeforeEach(func() {
+			namespaceNames := []string{"test-tenant", "test-tenant-2", "test-tenant-3", "test-tenant-4"}
```

Comment:
let's have `test-tenant-3` assigned to `namespace-access-3`, and `test-tenant-4` assigned to nothing. it is a bit confusing otherwise